lib/xandra/protocol.ex

```diff
@@ -18,6 +18,13 @@ defmodule Xandra.Protocol do
     %{frame | body: encode_string_map(requested_options)}
   end
 
+  def encode_request(%Frame{kind: :auth_response} = frame, requested_options, options)
+      when is_map(requested_options) do
+    [authenticator: authenticator, options: auth_options] = options[:authentication]
+    body = authenticator.response_body(auth_options)
+    %{frame | body: [<<byte_size(body)::32>>, body]}
```

Comment:
Since we'll use iodata for `body`, we should change `byte_size/1` here to `IO.iodata_length/1`.